src/main/java/cn/nukkit/block/BlockNetherWart.java

```diff
@@ -0,0 +1,99 @@
+package cn.nukkit.block;
+
+import cn.nukkit.Player;
+import cn.nukkit.Server;
+import cn.nukkit.event.block.BlockGrowEvent;
+import cn.nukkit.item.Item;
+import cn.nukkit.level.Level;
+import cn.nukkit.math.Vector3;
+import cn.nukkit.utils.BlockColor;
+
+import java.util.Random;
+
+/**
+ * Created by Leonidius20 on 22.03.17.
+ */
+public class BlockNetherWart extends BlockFlowable {
+
+    public BlockNetherWart(){
+        this(0);
+    }
+
+    public BlockNetherWart(int meta) {
+        super(meta);
+    }
+
+    @Override
+    public boolean place(Item item, Block block, Block target, int face, double fx, double fy, double fz) {
+        return this.place(item, block, target, face, fx, fy, fz, null);
+    }
+
+    @Override
+    public boolean place(Item item, Block block, Block target, int face, double fx, double fy, double fz, Player player) {
+        Block down = this.getSide(0);
+        if (down.getId() == SOUL_SAND) {
+            this.getLevel().setBlock(block, this, true, true);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public int onUpdate(int type) {
+        if (type == Level.BLOCK_UPDATE_NORMAL) {
+            if (this.getSide(Vector3.SIDE_DOWN).getId() != SOUL_SAND) {
+                this.getLevel().useBreakOn(this);
+                return Level.BLOCK_UPDATE_NORMAL;
+            }
+        } else if (type == Level.BLOCK_UPDATE_RANDOM) {
+            if (new Random().nextInt(10) == 1) {
+                if (this.meta < 0x02) {
+                    BlockNetherWart block = (BlockNetherWart) this.clone();
```

Comment:
It is used to send the old block state and the new block state to BlockGrowEvent